packages/shared/lib/db/migrations/20240222111223_data_records_index.cjs

```diff
@@ -0,0 +1,11 @@
+exports.config = { transaction: false };
+
+exports.up = function (knex) {
+    return knex.schema.raw(
+        'CREATE INDEX CONCURRENTLY "idx_records_created_id_connection_model" ON "nango"."_nango_sync_data_records" USING BTREE ("created_at","id","nango_connection_id","model");'
+    );
```

Comment:
it's reversed 
`idx_nango_records_composite`	=> `nango_connection_id, model, created_at, id`
My gut feeling is that the order is the slowest operation and an index starting with the order would improve things, not sure until it's applied. If it's useless it will just be discarded by the planner.

Reply:
To me the current index makes more sense.

According to https://www.postgresql.org/docs/current/indexes-multicolumn.html
```
A multicolumn B-tree index can be used with query conditions that involve any subset of the index's columns, but the index is most efficient when there are constraints on the leading (leftmost) columns. [...] Constraints on columns to the right of these columns are checked in the index, so they save visits to the table proper, but they do not reduce the portion of the index that has to be scanned. For example, given an index on (a, b, c) and a query condition WHERE a = 5 AND b >= 42 AND c < 77, the index would have to be scanned from the first entry with a = 5 and b = 42 up through the last entry with a = 5. Index entries with c >= 77 would be skipped, but they'd still have to be scanned through
```
Using a datetime as leftmost column seems less restrictive than connectionId/model.

Reply:
agreed but locally I still managed to make it pick up the index when we use the cursor pagination so I want to try